cylc/rose/utilities.py

```diff
@@ -999,3 +1003,94 @@ def copy_config_file(
+    --against-source argument is used in a Cylc script.
+    """
+    # if opts.against_source is a path then we are validating a source
+    # directory against installed options
+    rundir = opts.against_source
+
+    # If the calling script doesn't have this option (Everything except
+    # Cylc VR) then this defaults to false. If it's true we skip all
+    # following logic:
+    if not getattr(opts, 'clear_rose_install_opts', False):
+        opts.clear_rose_install_opts = False
+    else:
+        return opts
+
+    # NOTE: we only need to load the rose-suite-cylc-install for this
+    cli_config, _ = record_cylc_install_options(
+        srcdir, rundir, opts, modify=False
+    )
+
+    # Set ROSE_ORIG_HOST to ignored, and choose not to ignore it
+    # if this is a validation against source:
+    no_ignore = False
+    if rundir:
+        no_ignore = True
+        for keys, node in cli_config.walk():
+            if keys[-1] == 'ROSE_ORIG_HOST':
+                node.state = cli_config.STATE_SYST_IGNORED
+
+    # Get opt_conf_keys stored in rose-suite-cylc-install.conf
+    opt_conf_keys = cli_config.value.pop('opts').value.split(' ')
+    if any(opt_conf_keys):
+        opts.opt_conf_keys = opt_conf_keys
+
+    # Get --suite-defines/-S
+    # Work out whether user has used "template variables", "jinja2:suite.rc"
+    # or "empy:suite.rc" (There is an assumption that they aren't mixing
+    # them that is not guarded against):
+    for section in SECTIONS:
+        if cli_config.value.get(section, False):
+            template_variables = cli_config.value.pop(section)
+            break
+    # Create new -S list.
+    if any(template_variables):
+        opts.rose_template_vars = [
+            f'{keys[1]}={val.value}'
+            for keys, val in template_variables.walk(no_ignore=no_ignore)
+        ]
+
+    # Get all other keys (-D):
+    new_defines = []
+    for keys, value in cli_config.walk(no_ignore=no_ignore):
+        # Filter out section headings:
+        if not isinstance(value.value, dict):
+            section, key = keys
+            # Don't include section for top level items:
+            section = f"[{section}]" if section else ''
+            new_defines.append(f'{section}{key}={value.value}')
+
+    opts.defines = new_defines
+    return opts
+
+
+def sanitize_opts(opts):
+    """Warn if standard variables which will be overridden
+    have been added to CLI opts, and remove them.
+
+    n.b. This doesn't remove the need to check the input rose-suite.conf
+    for these variables.
+    """
+    options = []
+    for section in ['rose_template_vars', 'defines']:
+        if section in opts.__dict__ and opts.__dict__[section]:
+            for item in opts.__dict__[section]:
+                options.append((section, item))
+
+    for (section, item), (var_name, replace) in itertools.product(
+        options, STANDARD_VARS
+    ):
+        if re.match(rf'.*{var_name}=', item):
```

Comment:
With this change `ROSE_ORIG_HOST` will match, but `FOO_ROSE_ORIG_HOST` will not:

```suggestion
        if re.match(rf'.*\b{var_name}=', item):
```